### ISSUE TITLE
notifications for mentions that have been edited to no longer mention you look like generic "new post" notifications

### ISSUE THREAD

**Issue by codl** [tuskyapp/Tusky#3152]
if someone mentions you and then edits the post to no longer mention you, the notification stays (which i think makes sense) but changes to look like a "new post" notification (like when you've tapped the bell on someone's profile), with a "🏠 name just posted" header, a low contrast profile picture and text, and no buttons for reply, boost, fav, etc

I think this is misleading and could very easily lead to confusion

The mastodon web interface displays those like regular mentions, which is not ideal imo but still less confusing than turning them into a different type of notification

I think tusky should either do the same thing the web interface does, or, even better, add a "name mentioned you" header to the notification if the post doesn't visibly mention you

Before edit:
![a normal mention notification](https://user-images.githubusercontent.com/315139/211190671-ca0f7b4a-e9de-47fc-b06c-26af26e64310.jpg)
Same post and same notification, after edit:
![a "new post" notification](https://user-images.githubusercontent.com/315139/211190651-2b151d9c-c4c5-43ee-88d7-d5bb65aec1ba.jpg)

I had to close tusky between those screenshots so the notification would update

<details>
<summary>Response from /api/v1/notifications with a mention that has been edited to no longer be a mention.</summary>

````
[
  {
    "id": "2160090",
    "type": "mention",
    "created_at": "2023-01-08T10:10:20.582Z",
    "account": {
      "id": "19739",
      "username": "codl",
      "acct": "codl@donphan.social",
      "display_name": "codl 🎨",
      "locked": false,
      "bot": false,
      "discoverable": true,
      "group": false,
      "created_at": "2018-01-18T00:00:00.000Z",
      "note": "<p>good drawer</p><p>late 20s, they/them. </p><p>i occasionally draw smut (always under a CW)</p><p>hello!</p>",
      "url": "https://donphan.social/@codl",
      "avatar": "https://media.chitter.xyz/accounts/avatars/000/019/739/original/d31af8db90358626.png",
      "avatar_static": "https://media.chitter.xyz/accounts/avatars/000/019/739/original/d31af8db90358626.png",
      "header": "https://media.chitter.xyz/cache/accounts/headers/000/019/739/original/2579c8abef490e17.jpeg",
      "header_static": "https://media.chitter.xyz/cache/accounts/headers/000/019/739/original/2579c8abef490e17.jpeg",
      "followers_count": 347,
      "following_count": 50,
      "statuses_count": 741,
      "last_status_at": "2023-01-08",
      "emojis": [],
      "fields": [
        {
          "name": "main account",
          "value": "<span class=\"h-card\"><a href=\"https://chitter.xyz/@codl\" class=\"u-url mention\" rel=\"nofollow noopener noreferrer\" target=\"_blank\">@<span>codl@chitter.xyz</span></a></span>",
          "verified_at": null
        },
        {
          "name": "elsewhere",
          "value": "<a href=\"https://www.deviantart.com/codl\" rel=\"nofollow noopener noreferrer\" target=\"_blank\"><span class=\"invisible\">https://www.</span><span class=\"\">deviantart.com/codl</span><span class=\"invisible\"></span></a> <a href=\"https://codll.tumblr.com/\" rel=\"nofollow noopener noreferrer\" target=\"_blank\"><span class=\"invisible\">https://</span><span class=\"\">codll.tumblr.com/</span><span class=\"invisible\"></span></a> <a href=\"https://cohost.org/museumofcodl\" rel=\"nofollow noopener noreferrer\" target=\"_blank\"><span class=\"invisible\">https://</span><span class=\"\">cohost.org/museumofcodl</span><span class=\"invisible\"></span></a>",
          "verified_at": null
        }
      ]
    },
    "status": {
      "id": "109653040856691519",
      "created_at": "2023-01-08T10:10:20.000Z",
      "in_reply_to_id": null,
      "in_reply_to_account_id": null,
      "sensitive": false,
      "spoiler_text": "",
      "visibility": "direct",
      "language": "en",
      "uri": "https://donphan.social/users/codl/statuses/109653040839184950",
      "url": "https://donphan.social/@codl/109653040839184950",
      "replies_count": 0,
      "reblogs_count": 0,
      "favourites_count": 0,
      "edited_at": "2023-01-08T10:11:22.000Z",
      "favourited": false,
      "reblogged": false,
      "muted": false,
      "bookmarked": false,
      "content": "<p>actually i have changed my mind and no longer want to talk to you. i have edited out the mention</p>",
      "reblog": null,
      "account": {
        "id": "19739",
        "username": "codl",
        "acct": "codl@donphan.social",
        "display_name": "codl 🎨",
        "locked": false,
        "bot": false,
        "discoverable": true,
        "group": false,
        "created_at": "2018-01-18T00:00:00.000Z",
        "note": "<p>good drawer</p><p>late 20s, they/them. </p><p>i occasionally draw smut (always under a CW)</p><p>hello!</p>",
        "url": "https://donphan.social/@codl",
        "avatar": "https://media.chitter.xyz/accounts/avatars/000/019/739/original/d31af8db90358626.png",
        "avatar_static": "https://media.chitter.xyz/accounts/avatars/000/019/739/original/d31af8db90358626.png",
        "header": "https://media.chitter.xyz/cache/accounts/headers/000/019/739/original/2579c8abef490e17.jpeg",
        "header_static": "https://media.chitter.xyz/cache/accounts/headers/000/019/739/original/2579c8abef490e17.jpeg",
        "followers_count": 347,
        "following_count": 50,
        "statuses_count": 741,
        "last_status_at": "2023-01-08",
        "emojis": [],
        "fields": [
          {
            "name": "main account",
            "value": "<span class=\"h-card\"><a href=\"https://chitter.xyz/@codl\" class=\"u-url mention\" rel=\"nofollow noopener noreferrer\" target=\"_blank\">@<span>codl@chitter.xyz</span></a></span>",
            "verified_at": null
          },
          {
            "name": "elsewhere",
            "value": "<a href=\"https://www.deviantart.com/codl\" rel=\"nofollow noopener noreferrer\" target=\"_blank\"><span class=\"invisible\">https://www.</span><span class=\"\">deviantart.com/codl</span><span class=\"invisible\"></span></a> <a href=\"https://codll.tumblr.com/\" rel=\"nofollow noopener noreferrer\" target=\"_blank\"><span class=\"invisible\">https://</span><span class=\"\">codll.tumblr.com/</span><span class=\"invisible\"></span></a> <a href=\"https://cohost.org/museumofcodl\" rel=\"nofollow noopener noreferrer\" target=\"_blank\"><span class=\"invisible\">https://</span><span class=\"\">cohost.org/museumofcodl</span><span class=\"invisible\"></span></a>",
            "verified_at": null
          }
        ]
      },
      "media_attachments": [],
      "mentions": [],
      "tags": [],
      "emojis": [],
      "card": null,
      "poll": null
    }
  }
]
````
</details>

TLDR: the notification's `type` is still `mention` but my account is not in `status.mentions`

* * * *
- Tusky Version: Tusky Test 20.0-0328c4e8

**Comment by connyduck** [comment]
Can't reproduce anymore. Also there is nothing in the code that could explain this, the only thing determining how the post is rendered is the `type`.

**Comment by codl** [comment]
can't reproduce either :)
